src/csexp_rpc/csexp_rpc.ml

```diff
@@ -284,7 +284,7 @@ module Client = struct
           let in_ = Unix.in_channel_of_descr client in
           (in_, out))
     in
-    Session.create in_ out
+    Session.create_full Socket in_ out
```

Comment:
From what I can tell, the effect of this change is to replace "close" with "shutdown".

For the affected fd, did we have double-close before this change, or do we have 0 closes after this change?

If we had double-close, I claim it's a bug that that wasn't reported. If we have 0 closes, it's also a bug (an fd leak). Either way, that seems to suggest we may have a bug remaining?